### PR TITLE
Fix of #282

### PR DESCRIPTION
**Pull request by dongbohu**
I found the issue #282 is important but easy to fix. Here it is.

I also make the timeout value of the slider rendering longer. The original value (`250 ms`) seems to work fine on my desktop, but not on the production server. `1000ms` should make it better.

### CODE REVIEW

**Comment by dongbohu** [comment]
Here is the demo page:
http://165.123.67.202/#/gene_network/?genes=6552&mlmodel=4
The signature names have also been padded in this demo server.

**Comment by rzelayafavila** [comment]
@dongbohu do you have this running in a test instance to check out?

File: interface/src/app/gene/network/network.js [hunk 1]
```diff
@@ -157,9 +157,9 @@ angular.module('adage.gene.network', [
       var refreshSlider = function() {
         $timeout(function() {
           $scope.$broadcast('rzSliderForceRender');
-        }, 250);
-        // Note by dhu: 250ms is an arbitrary timeout value. The minimum timeout
-        // value that worked on my desktop is 150ms.
+        }, 1000);
+        // Note by dhu: 1000ms is an arbitrary timeout value. The minimum
+        // timeout value that worked on my desktop is 150ms.
```
Comment (mhuyck) [inline interface/src/app/gene/network/network.js]: This fixes the slider glitch for me, too. Nice!

The fact that it works for a longer timeout but not for a shorter one is an indication that there is a race condition here. Some event must happen before issuing the `rzSliderForceRender` and it happens sometimes within 250ms and sometimes not. If we knew what that event was we would be able to make a 100% reliable fix. If we don't know what the event is, then this solution is a good enough workaround for me.

File: interface/src/app/gene/network/network.js [hunk 2]
```diff
@@ -327,7 +327,10 @@ angular.module('adage.gene.network', [
           var heavyGenes = [data.gene1.id, data.gene2.id].join(',');
           var target = d3.event.target;
           Signature.get(
-            {'heavy_genes': heavyGenes, 'limit': 0},
+            {'heavy_genes': heavyGenes,
+              'mlmodel': MlModelTracker.id,
```
Comment (mhuyck) [inline interface/src/app/gene/network/network.js]: Nice and simple to fix. Great!

File: interface/src/app/gene/network/network.js (hunk 1 above)
Comment (rzelayafavila) [reply interface/src/app/gene/network/network.js]: :+1:

File: interface/src/app/gene/network/network.js (hunk 2 above)
Comment (rzelayafavila) [reply interface/src/app/gene/network/network.js]: :+1:

**Comment by mhuyck** [comment]
Just had one other thought, but it should not hold up merging this fix: can you make a test case that would have caught this problem?

**Comment by dongbohu** [comment]
I took a look at the back end unit test before submitting this PR. There is actually one test case for this `heavy_genes` filter. But to catch this bug, we really need a front end E2E test to simulate the model selecting and edge clicking behavior.

**Comment by mhuyck** [comment]
That's true. Should we make an issue to follow up after we get E2E testing working? I think we will have a much easier time with E2E tests after we get Docker deployment finalized. (The issue I recall having with the E2E setup was specific to our AWS EC2 deployment.)

File: interface/src/app/gene/network/network.js (hunk 1 above)
Comment (dongbohu) [reply interface/src/app/gene/network/network.js]: @mhuyck What you said about race condition is a good point. I took a look at the source code of `rzslider`. Unsurprisingly the event is related to a lot of DOM manipulations. I tried a few quick fixes but they didn't work. The fact that its README.md suggests using `$timeout()` probably indicates that that is the best solution.

I will add a new issue to keep track of it though.

**Comment by dongbohu** [comment]
@mhuyck: I just talked to @casey about the importance of having E2E test on the front end too. I will add an issue for now.
